### DISCUSSION TITLE
Version 1.1 Build 1

### DISCUSSION THREAD

**Discussion by ErinaSugino** [ErinaSugino/Starbound-SBR-Mod#44] [Update Builds]
## Version 1.1 Patchnotes
- Integrated Lustbound into SBR
- Added modular pregnancy management (Including multi-baby pregnancies)
- Added new cumflation and fertility stacking mechanics (Thanks to XspeedPL)
- Added properly working baby genetics
- Re-added global NPC auto-climax
- Outsourced UI sync data for easy patching
- Added stillborn chance

## Build 1 Notes
- Build is not tested. This pre-release is for public testing purposes only.

### Important Notes
- Do not use on main savefile - 1.1 changes the data format for pregnancies which is not backwards compatible. Reverting back to 1.0 will erase or break all ongoing pregnancies!

<hr /><em>This discussion was created from the release <a href='https://github.com/ErinaSugino/Starbound-SBR-Mod/releases/tag/v1.1_b1'>Version 1.1 Build 1</a>.</em>

**Comment by Mixarar** [comment]


**Reply by ErinaSugino** [reply]
That is correct. Core Lustbound has been merged into this build, as such you neither need to install `Lustbound Base`, `Lustbound Base Manager` or `SBR Lustbound Patch` anymore, while it still works with the other original Lustbound addons for, e.g., Aphrodite's Bow or Omnibrowser.

If it doesn't, that classifies as a bug.

**Reply by ErinaSugino** [reply]
Fixed that one, uploaded new zip for build 1.

Furthermore, keep in mind that early stage testing is most easily done on a clean setup where you log isn't natively spammed with a million error messages from other, fully unrelated mods, to track actual bugs in V1.1.

Also finally, I hope you have made a backup of your savefile beforehand. 1.1 changes the structure of pregnancy data, even in this build, and switching back to an older version will either erase or break all ongoing pregnancies.

**Reply by ErinaSugino** [reply]
I know you pointed out the error line, but - and this is seriously not meant to be offensive - I still have to look at the rest too. Lua errors can be incredibly finicky and error messages can be misleading as hell.

Regarding the new error:
Open `scripts/sexbound/lib/sexbound/actor.lua`, go to line 1368 and replace
```lua
frameName       = self:getFrameName(actor:getAnimationState()),
```
with
```lua
frameName       = self:getFrameName(self:getAnimationState()),
```

I don't want to spam the repo with a million one-line fixes and I expect more of this kind of stuff to pop up for the next eternity (this is what testing is for after all), so I'll hold back on pushing the changes and reuploading the build.

**Reply by ErinaSugino** [reply]
`scripts/sexbound/plugins/climax/climax.lua` in line 402
```lua
local _actor = actor:getParent():getActors()[target] or nil
```
to
```lua
local _actor = self:getParent():getParent():getActors()[target] or nil
```

<hr>

`scripts/sexbound/lib/sexbound/actor.lua` in line 1376
```lua
isSwollen           = self:isBellySwollen(),
```
to
```lua
isInflated           = self:isInflated(),
```

**Reply by ErinaSugino** [reply]
`scripts/sexbound/plugins/climax/climax.lua` in line 103
```lua
self:inflate(message)
```
to
```lua
self:inflate(message:getData())
```

Regarding the dialogue - I don't see any specific error from that module, so I'mma gonna assume for now it's a byproduct of a core script erroring on update, i.e. every tick. You also said the dialogue issues started arising after the climax problem, so that lines up.

**Reply by ErinaSugino** [reply]
Oh, a proper bug for the first time. Guess I'm done here for this batch then, I'll publish the changes so far and look into it tomorrow.

The menu closing not removing your character from the animation is a bug, but since you didn'T attach a log this time I have no idea where it might be.

You mean standing? That's a known issue, and will be fixed once red finished the new positions. One major reason for the big changes in position management and filtering based on body traits was that red wanted this option to make more scenario-specificly tailored new positions, that *don't* look that out-of-place. Including splitting standing into a dedicated twoactor and a dedicated threeactor version with different requirements and potentially even different sprites. So I didn't change anything about that.

<hr>

If a godly tool that handy existed, we Starbound modders woudn't age by 10 years per workday. Sadly, I don't know of anything you can conveniently hook into Starbound to get proper debugger functionality. And building something like this ourselves *in* Lua ist - stupid. In order to debug errors, *your debugger can't be on the same level that'S erroring, hence breaking the execution loop*. Lua itself, on a global level, has a "debug" library with more detailed functions to find errors manually - but this library is disabled in Starbound for safety reasons. It gives more power than a modder should have over your PC. You'd need to enable "unsafeMods" by manually editing your starbound config - but **for the love of god and I am serious here never ever do that unless you only have the SINGLE one mod installed you want to debug, and you FULLY trust it's developer, or it's Open Source and someone verified it's safety!**

**Reply by ErinaSugino** [reply]
The inflation thingie should need 7.5 by default, and only decrease during idle. I can add some random console outputs for the next build the check that.

The logging settings are literally at the bottom of the main config file, second to last option.
```json
  /**
   * Logging level. All log data placed into the player's starbound.log file.
   */
  "log": {
    "showDebug": false,
    "showInfo": false,
    "showWarn": true,
    "showError": true
  },
```

About the error...
...are you using a custom race in that moment?

**Reply by ErinaSugino** [reply]
... try removing StarExtension.

**Reply by ErinaSugino** [reply]
Are you saying you removed StarExtension while the game was running? Oof. That one's on you.

Ok, yes, cool, cumflation is broken in every way possible. Got it. I'll reopen the Issue and you can try to explain things there in great detail and all.

Regarding pregnancies - info/debug logging on, log goes here. Even if it doesn't say "ERROR" or "WARN", there can still be lot of error logging in there.

And finally - ...what?

**Reply by ErinaSugino** [reply]
All poses are supposed to make particles. The only thing capable of intendedly stopping that is the condom. But that would also prevent the liquid from spawning.

**Reply by ErinaSugino** [reply]
Huh. Guess Loc decided there shall be no particles for actor 1 in cowgirl. Guess I'll have to find out why.

And the whole apparel stuff has always been weird. Might be nice, very subtle, one pixel large touch for people wanting extra immersion but... Fixing this has no priority whatsoever.

**Reply by ErinaSugino** [reply]
Step 1: Edit the .particle file
Step 2: ???
Step 3: Profit

**Comment by XspeedPL** [comment]
Oh, the mod is at last working again after [b2bccaf](https://github.com/ErinaSugino/Starbound-SBR-Mod/commit/b2bccaf93235012afe8d5b8a5f125f7365111c42), nice work

**Comment by Mixarar** [comment]


**Reply by ErinaSugino** [reply]
1. That shouldn't be. Looks misplaced. Needs investigation.
2. Yes
3. It supports it, the system is there (albeit untested of course). Other than the default "baby" (which is what was in SxB before all this time) I won't create any others. The system exists so that other modders can create custom stuff for their mods - e.g. Teratophilia. Sexbound itself does not offer any features for monsters. Terato is a mod designed for just that. The creator of Terato has been asking me about the possibility to make different pregnancy types, like egg-based ones, for monsters. Here is the possibility.
4. Planned is they don't look like crap. But that's not my fight to fight. I'm a coder, not an artist. Just like with additional positions, better sprites or maybe even anims are what red said he's working on (along his other, like, 5 projects. Man's busy, give him a hand.) Would custom race mods need to update? Yes, if they want their own idle sprites. No otherwise.
5. The birthing thing is currently just forcing the player into a solo sexnode with only one custom position with is only the one laying frame of the missionary actor 2. If someone wants to create an actual animation for that, go ahead. It is currently in this build.
6. You can take a look at the full changelog - while cropped here on GitHub due to... concerns regarding me publicly announcing adding true kids to a sexmod, the full version with the "planned features" section and ideas section is in every release.
7. At that point I'd probably just spend more time maintaining and working off of other's ideas, so no. What is currently, thematically, being worked on is in the roadmap for the current version, or somewhat marked in the full changelog. Sometimes I make my own Issues to keep track of things, too.
8. They will be exactly like a regular NPCs but smaller. And obviously not sexually interactable. If I figure out custom vanilla dialogue, they shall also get specific kid-themed dialogue.
9. See your other thread.
10. Whenever this mess is fixed and the next big feature is implemented.
11. With Bumple gone that makes things complicated, as the mod being highly used and abandoned was the whole reason to assimilate Lustbound in the first place, and I personally very much like the concept of the mod (and it's last released beta did have the basics down and working iirc), also there are some stupid design flaws I couldn't talk Bumple out of which I'd need to patch anyway - but then again, at some point if I just keep inhaling other mods things become too big and out of hand. I don't want to become the next FU. It also goes against the core concept of both mods - Sexbound is generally the core API enabling sexual content in Starbound, designed so that other people can create mods for it to add stuff that users can then opt to use or not. MOTH was designed a standalone dating sim, with Sexbound integration only planned for *later down the road* because "You can make babies anyway, so why not show the process?". The two mods are fundamentally too far aparts, so no, no immediate plans to merge that one.<br><br>Most basic "family relation" is supposed to be a thing for 1.1 though, that is as far as keeping track of 2 levels of incest goes. With every NPC keeping track of their father and mother UUID to find out if they're related by one or two layers. That, however, required me to first figure out how to edit behaviour trees to use this data and how to alter dialogue per target entity, to play family-themed dialogue only for family members.
12. Not. Monsters don't use any of this data. Unless of course Terato adds it in.
13. Even persistent NPCs that don't despawn only "exists" as long as they are loaded in, which is only the case when a player is in the same world and the NPC is basically on-screen. Otherwise the NPC is unloaded, and no scripts execute, so no stats update.
14. Not. You can't do jack shit to entities. You could kill and respawn a clone of an NPCs with different parameters - but the player is untouchable. See your other thread for a more detailed explanation.

**Reply by ErinaSugino** [reply]
Oh whoops, put a wrong template in the config. The innermost object needs to be an array.

**Reply by ErinaSugino** [reply]
No, I said Array instead of Object. [] instead of {}

**Reply by ErinaSugino** [reply]
Yes, that's how it works

**Reply by ErinaSugino** [reply]
The child species? Pregnancy becomes impossible. Only add what exists, that's rule number 1 in Starbound modding.

**Reply by ErinaSugino** [reply]
The genetics table is completely new, in this preview build even. Nobody could have added stuff there yet. Also, as long as the \*child\* species adds itself to the list, everything is fine.

**Comment by Mixarar** [comment]


**Reply by ErinaSugino** [reply]
Possible to upload? Yes. Is it a death sentence? Also yes. No matter what tags are available, or what they say, Steam hates adult content. Not to mention that workshops are usually moderated by people from the developer/publisher, and I don't think Chucklefish likes sex in Starbound. So I certainly wouldn't risk my ass for it. People on Steam already know of the good Sauce anyway.

Secondly - "Where it wouldn't require having configs outside of the game". What? This makes no sense whatsoever.

**Reply by ErinaSugino** [reply]
They would be there even if the mod was on Steam. Only difference is, Steam only allows .pak mods, so noone would ever be able to configure the mod unless they learn how to read content IDs, manually un-pak the mod from their Steam workshop folder, and bypass Steam's automatic integritiy check that would restore the mod to it's original state. In other words: Everything would be worse on every level if the mod was on Steam.

**Reply by ErinaSugino** [reply]
That is already known, and about 3 different Discord servers are monitoring, laughing and rioting about the situation. That dude did also upload SBR, but I got it taken down.

Regarding the pregnancy issue, pregnancy has nothing to do with the cumflation mechanic and as such is it's own issue, yes.

**Comment by ErinaSugino** [comment]
Published a new zip with the current build that includes the changes from d22e795
Mostly just includes means to actually find the errors rather than fixes.

**Reply by ErinaSugino** [reply]
New zip with changes from e40b976
Fixes immediate pregnancy error, allowing for the new mechanic to be properly tested.

**Reply by ErinaSugino** [reply]
New zip with changes from fcd8b5c
Fixes pregnancies (tested myself with two normal humans)
